python/cudf/cudf/core/column/column.py

```diff
@@ -1652,28 +1652,26 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
                     sr = pd.Series(arbitrary, dtype="str")
                     data = as_column(sr, nan_as_null=nan_as_null)
                 else:
-                    try:
-                        tmp_data = np.asarray(
+                    if dtype is None and pd.api.types.infer_dtype(
+                        arbitrary
+                    ) in ("mixed", "mixed-integer"):
+                        data = np.asarray(arbitrary, dtype="object")
+                        try:
+                            data = as_column(
+                                data, dtype=dtype, nan_as_null=nan_as_null
+                            )
```

Comment:
What's the situation where it's `mixed` or `mixed-integer` and it gets properly turned into a native type?

Reply:
Its is take care by arrow, such as converting cupy scalars and other options.